### PR TITLE
[Release 2.16.0] Consolidated release notes

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
[Release 2.16.0] Consolidated release notes 

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/4771

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.16.0.md [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+* Application-based templates have been added, providing default settings that can simplify the work of tuning your indices for performance of compute and storage resources as well as for usability for common use cases.
+* Support for multiple data sources is extended to two more Dashboards plugins — Notebooks and Snapshot — and all plugins now have version decoupling support in place to filter out incompatible data sources from the selection.
+* In February 2024, OpenSearch issued a deprecation notice regarding CentOS Linux 7, which reached end-of-life on June 30, 2024. As of this release, OpenSearch is deprecating CentOS Linux 7 as a continuous integration build image and supported operating system.
+* The Query Insights plugin is now bundled as a default plugin in the OpenSearch 2.16 distribution. You can use the top n queries APIto identify rogue queries more easily.
+
+### EXPERIMENTAL FEATURES
```
Comment (prudhvigodithi) [inline release-notes/opensearch-release-notes-2.16.0.md:19]: Still Pending on adding them, will sync with @jhmcintyre.

Comment (prudhvigodithi) [reply release-notes/opensearch-release-notes-2.16.0.md:19]: Just added.

Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.16.0.md:19]: @prudhvigodithi  wanted to flag the release highlights are not yet final, pending editorial review. Aiming for completion on this Tuesday morning.

File: release-notes/opensearch-release-notes-2.16.0.md [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+## FEATURES/ENHANCEMENTS
+
+
+### Opensearch Dashboards Notifications
+
+
+* side navigation changes for notifications ([#222](https://github.com/opensearch-project/dashboards-notifications/pull/222))([#225](https://github.com/opensearch-project/dashboards-notifications/pull/225))
+* Backport/backporting to 2.x ([#223](https://github.com/opensearch-project/dashboards-notifications/pull/223))
+
+
```
Comment (peterzhuamazon) [inline release-notes/opensearch-release-notes-2.16.0.md:874]: 1. Need to talk to the OSD Notification team on their formatting being wrong, should separate feature and enhancement to two different categories.
2. Confirm with them then put the section to the corresponding category.

Thanks.

Comment (prudhvigodithi) [reply release-notes/opensearch-release-notes-2.16.0.md:874]: Sure @peterzhuamazon.

**Comment by prudhvigodithi** [comment]
Adding @jhmcintyre @hdhalter.

**Review by jhmcintyre — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**